exercises/queen-attack/example.exs

```diff
@@ -38,6 +39,15 @@ defmodule Queens do
     white_x == black_x || white_y == black_y || diagonal?(white, black)
   end
 
+  defp is_valid_position?({x, y}) do
+    x in @board_range && y in @board_range
+  end
+
+  defp add_to_queens(queens, {queen, pos}) do
+    if pos in Map.values(queens), do: raise(ArgumentError)
+    %{queens | queen => pos}
+  end
```

Comment:
I am unsure how I feel about any duplicate raising an error.  I think what I'd rather see is is a check for each queen

```elixir
case queen do
  :black -> if queens.white == pos, do: raise ArgumentError, "white queen already placed here"
  :white -> if queens.black == pos, do: raise ArugmentError, "black queen already placed here"
  _ -> raise ArguementError, "invalid queen color"
end
```

I'm not sure if it matters if they try to redefine themselves.